.github/workflows/test.yml

```diff
@@ -0,0 +1,33 @@
+on: [push]
+
+jobs:
+  test-job:
+    runs-on: ubuntu-latest
+    name: Run script test
+    steps:
+      - uses: actions/checkout@v4.1.1
```

Comment:
Here you are not fetching the entire history: `git log` returns just the last commit.

Reply:
Here we need only the last commit with action last version code. The deep fetch is done inside the action code, where `fetch-depth: 0` is used.